src/components/auth/LoginWithAuthServerPage.tsx

```diff
@@ -1,41 +1,71 @@
+import { ChainId, ProviderType } from '@dcl/schemas'
 import { Button } from 'decentraland-ui/dist/components/Button/Button'
-import { AuthServerProvider, connection } from 'decentraland-connect'
-import { authenticate } from '../../kernel-loader'
+import { WearablePreview } from 'decentraland-ui/dist/components/WearablePreview/WearablePreview'
+import { Loader } from 'decentraland-ui/dist/components/Loader/Loader'
+import { AuthServerProvider } from 'decentraland-connect'
+import { authenticate, getWantedChainId } from '../../kernel-loader'
 import './LoginWithAuthServerPage.css'
 
 enum View {
+  LOADING,
   WELCOME,
+  WELCOME_CONNECTED,
   SIGN_IN_CODE,
   EXPIRED
 }
 
 export const LoginWithAuthServerPage = () => {
-  const [view, setView] = useState<View>(View.WELCOME)
+  const [view, setView] = useState<View>(View.LOADING)
   const [disabled, setDisabled] = useState(false)
 
   const initSignInResultRef = useRef<any>()
   const expirationTimeoutRef = useRef<NodeJS.Timeout>()
+  const connectedAccountRef = useRef<string>()
+  const connectedNameRef = useRef<string>()
 
   useEffect(() => {
-    const url = new URL(window.location.href)
-
-    switch (url.origin) {
-      case 'https://decentraland.org':
-      case 'https://play.decentraland.org':
-        AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.org')
-        AuthServerProvider.setAuthDappUrl('https://decentraland.org/auth')
-        break
-      case 'https://decentraland.today':
-      case 'https://play.decentraland.today':
-        AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.today')
-        AuthServerProvider.setAuthDappUrl('https://decentraland.today/auth')
-        break
-      default:
-        AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.zone')
-        AuthServerProvider.setAuthDappUrl('https://decentraland.zone/auth')
-    }
+    ;(async () => {
+      const url = new URL(window.location.href)
+
+      switch (url.origin) {
+        case 'https://decentraland.org':
+        case 'https://play.decentraland.org':
+          AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.org')
+          AuthServerProvider.setAuthDappUrl('https://decentraland.org/auth')
+          break
+        case 'https://decentraland.today':
+        case 'https://play.decentraland.today':
+          AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.today')
+          AuthServerProvider.setAuthDappUrl('https://decentraland.today/auth')
+          break
+        default:
+          AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.zone')
+          AuthServerProvider.setAuthDappUrl('https://decentraland.zone/auth')
+      }
+
+      const connectedAccount = AuthServerProvider.getAccount()
+
+      if (connectedAccount) {
+        connectedAccountRef.current = connectedAccount
+
+        const env = getWantedChainId() === ChainId.ETHEREUM_SEPOLIA ? 'zone' : 'org'
```

Comment:
```suggestion
        const env = getWantedChainId() === ChainId.ETHEREUM_MAINNET ? 'org' : 'zone'
```
Can we do this so if we ever have to migrate from Sepolia to a new one we don't have to update this?